libcfnet/protocol.c

```diff
@@ -88,3 +89,93 @@ Seq *ProtocolOpenDir(AgentConnection *conn, const char *path)
 
     return seq;
 }
+
+int ProtocolStat(AgentConnection *conn, const char *remote_path,
+                 struct stat *stat_buf)
+{
+    assert(conn != NULL);
+    assert(remote_path != NULL);
+    assert(stat_buf != NULL);
+
+    time_t tloc = time(NULL);
+    if (tloc == (time_t) -1)
+    {
+        Log(LOG_LEVEL_ERR, "Couldn't read system clock (time: %s)",
+            GetErrorStr());
+        tloc = 0;
+    }
+
+    char buf[CF_BUFSIZE] = {0};
+    // Won't this cause problems on 32-bit systems? time_t would always be 64-bit, right?
```

Comment:
I don't think this is a safe assumption.

Reply:
It isn't. And `time_t` should be safe until at least 2038 :^)